packages/flutter_tools/lib/src/commands/upgrade.dart

```diff
@@ -68,16 +69,22 @@ class UpgradeCommand extends FlutterCommand {
     printStatus('');
     printStatus(flutterVersion.toString());
 
-    final String projRoot = findProjectRoot();
-    if (projRoot != null) {
+    final String projectRoot = findProjectRoot();
+    if (projectRoot != null) {
       printStatus('');
-      await pubGet(context: PubContext.pubUpgrade, directory: projRoot, upgrade: true, checkLastModified: false);
+      await pubGet(context: PubContext.pubUpgrade, directory: projectRoot, upgrade: true, checkLastModified: false);
     }
 
     // Run a doctor check in case system requirements have changed.
     printStatus('');
     printStatus('Running flutter doctor...');
-    await doctor.diagnose();
+    code = await runCommandAndStreamOutput(
+      <String>[
+        fs.path.join(Cache.flutterRoot, 'bin', 'flutter'), 'doctor',
```

Comment:
Nit: if the working directory is already `Cache.flutterRoot`, then you can omit `Cache.flutterRoot` here.